documentation/docs/15-types.md

```diff
@@ -74,3 +74,66 @@ export async function get({ params }) {
 > For this to work, your own `tsconfig.json` or `jsconfig.json` should extend from the generated `.svelte-kit/tsconfig.json` (where `.svelte-kit` is your [`outDir`](/docs/configuration#outdir)):
 >
 >     { "extends": "./.svelte-kit/tsconfig.json" }
+
+#### Default tsconfig.json
+
+The generated `.svelte-kit/tsconfig.json` file contains a mixture of options. Some are generated programmatically based on your project configuration, and should generally not be overridden without good reason:
+
+```json
+/// file: .svelte-kit/tsconfig.json
+{
+	"compilerOptions": {
+		"baseUrl": "..",
+		"paths": {
+			"$lib": "src/lib",
+			"$lib/*": "src/lib/*"
+		},
+		"rootDirs": ["..", "./types"]
+	},
+	"include": ["../src/**/*.js", "../src/**/*.ts", "../src/**/*.svelte"],
+	"exclude": ["../node_modules/**", "./**"]
+}
+```
+
+Others are required for SvelteKit to work properly, and should also be left untouched unless you know what you're doing:
+
+```json
+/// file: .svelte-kit/tsconfig.json
+{
+	"compilerOptions": {
+		// you must use `import type` to
+		// import types, rather than `import`,
+		// for svelte-preprocess to work
```

Comment:
The last line isn't 100% correct, it's more a hint like "if all of the imports are used as types, throw an error". Writing this, I'm wondering how this plays out in conjunction with "preserveValueImports".. maybe we can remove this option. Need to test this out, don't know when I will get to that.

Btw since TS 4.5 you can mix value and type imports by doing `import { type Foo, bar } ..`.

Reply:
I updated it — it's slightly tricky because we need to communicate how it affects how people should write code, more than what the option actually does, but it also needs to make sense in context